Actions/CheckForUpdates/CheckForUpdates.ps1

```diff
@@ -3,363 +3,161 @@
 
-    # Build the $archiceUrl instead of using the GitHub API
-    # The GitHub API has a rate limit of 60 requests per hour, which is not enough for a large number of repositories using AL-Go
-    $archiveUrl = "$($templateUrl -replace "https://www.github.com/","$ENV:GITHUB_API_URL/repos/" -replace "https://github.com/","$ENV:GITHUB_API_URL/repos/")/zipball/$templateBranch"
+# If templateUrl has changed, download latest version of the template repository (ignore templateSha)
+if ($repoSettings.templateUrl -ne $templateUrl -or $templateSha -eq '') {
+    $downloadLatest = $true
+}
 
-    Write-Host "Using template from $templateUrl@$templateBranch"
-    Write-Host "Using ArchiveUrl $archiveUrl"
+$templateFolder = DownloadTemplateRepository -headers $headers -templateUrl $templateUrl -templateSha ([ref]$templateSha) -downloadLatest $downloadLatest
+Write-Host "Template Folder: $templateFolder"
 
-    # Download the template repository and unpack to a temp folder
-    $headers = @{
-        "Accept" = "application/vnd.github.baptiste-preview+json"
-        "token" = $token
-    }
-    $tempName = Join-Path ([System.IO.Path]::GetTempPath()) ([Guid]::NewGuid().ToString())
-    InvokeWebRequest -Headers $headers -Uri $archiveUrl -OutFile "$tempName.zip" -retry
-    Expand-7zipArchive -Path "$tempName.zip" -DestinationPath $tempName
-    Remove-Item -Path "$tempName.zip"
+$templateBranch = $templateUrl.Split('@')[1]
+$templateOwner = $templateUrl.Split('/')[3]
 
-    # CheckFiles is an array of hashtables with the following properties:
-    # dstPath: The path to the file in the current repository
-    # srcPath: The path to the file in the template repository
-    # pattern: The pattern to use when searching for files in the template repository
-    # type: The type of file (script, workflow, releasenotes)
-    # The files currently checked are:
-    # - All files in .github/workflows
-    # - All files in .github that ends with .copy.md
-    # - All PowerShell scripts in .AL-Go folders (all projects)
-    $srcGitHubPath = '.github'
-    $srcALGoPath = '.AL-Go'
-    if ($directALGo) {
-        # When using a direct link to an AL-Go repository, the files are in a subfolder of the template repository
-        $typePath = $repoSettings.type
-        if ($typePath -eq "PTE") {
-            $typePath = "Per Tenant Extension"
+if (-not (IsDirectALGo -templateUrl $templateUrl)) {
+    $ALGoSettingsFile = Join-Path $templateFolder "*/.github/AL-Go-Settings.json"
+    if (Test-Path -Path $ALGoSettingsFile -PathType Leaf) {
+        $templateRepoSettings = Get-Content $ALGoSettingsFile -Encoding UTF8 | ConvertFrom-Json | ConvertTo-HashTable -Recurse
+        if ($templateRepoSettings.Keys -contains "templateUrl" -and $templateRepoSettings.templateUrl -ne $templateUrl) {
+            throw "The specified template repository is not a template repository, but instead another AL-Go repository. This is not supported."
```

Comment:
This test is that if the templateUrl in the template specified by the URL is not the same as the template specified by the URL itself - then the URL is pointing to an AL-Go repository instead of an AL-Go template.
Like - run Update AL-Go System Files with microsoft/bcsamples-bingmaps.pte - then you will get this error.